resources/js/components/WbsIssues.vue

```diff
@@ -79,45 +78,34 @@
     },
 
     methods: {
-      newSibling(ofIssueIndex) {
+      newNode(ofIssueIndex, parentId, indentation = 0) {
         const ofIssue = this.issues[ofIssueIndex];
 
         if ( ! ofIssue.hasOwnProperty('id')) {
           return;
         }
 
-        let targetIndex = ofIssueIndex + 1;
+        const targetLevel = ofIssue.level + indentation;
+        const targetIndex = this.getTargetIndex(ofIssueIndex, targetLevel);
 
-        for (let i = targetIndex; i < this.issues.length; i++) {
-          if (this.issues[i].level < ofIssue.level) {
-            break;
-          }
-
-          targetIndex = i;
-        }
-
-        this.newIssue(targetIndex + 1, ofIssue.parent_id, ofIssue.level);
+        this.insertNote(targetIndex, parentId, targetLevel);
       },
 
-      newChild(ofIssueIndex) {
-        const ofIssue = this.issues[ofIssueIndex];
+      getTargetIndex(fromIndex, minLevel) {
```

Comment:
If you move the `+ 1` outside of this method, I think it could be simplified to be more readable:

```
let targetIndex = fromIndex;

for (; targetIndex < this.issues.length; targetIndex++) {
    if (this.issues[i].level < minLevel) {
        break;
    }
}

return targetIndex;
```